src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -167,4 +168,34 @@ public function __invoke()
             );
         });
     }
+
+    /**
+     * @param string|null $region
+     *
+     * @return StsClient
+     */
+    private function createDefaultStsClient(
```

Comment:
I think this method is redundant. We can use the one defined in `CredentialProvider`. Just change the type of `$credentials` to be `CredentialsInterface|bool` and when called from `AssumeRoleWithWebIdentityCredentialProvider` just pass false as the credentials parameter.